docs/workforce-management-framework/topics/con-security-introduction.adoc

```diff
@@ -0,0 +1,49 @@
+[]
+= Introduction to {WFM-RC-NameShort} security features
+
+{WFM-RC-NameShort} server gives developers a choice one of the following authentication and authorisation providers
+link:http://www.keycloak.org/[Keycloak] and link:http://passportjs.org/[Passport.js].
+{WFM-RC-NameShort} is build around security interface, which abstracts from any security libraries.
+Both implementations can be swapped at any time without breaking internal api of {WFM-RC-NameShort} modules and reference applications.
+Users can also provide their own security implementation by implementing their own security interface.s
+
+Another security measures are incorporated into our reference applications.
+Please note that all link:https://expressjs.com/[express] based routes outside WFM modules will also need to use these security interfaces.
+
+== How to apply authorization using security interface
+You can choose Keycloak or Passport.js implementation but you can securing endpoints in the same way.
+You need to import link:++../../../api/{WFM-RC-Api-Version}/auth-passport/docs/interfaces/_src_auth_passportauth_.endpointsecurity.html++[EndpointSecurity]
+interface from passportAuth. Apply either implementation initialisation to the interface.
+You can then use this security interface protect method to guard routes with roles.
+
+== Passport.js
+Passport.js is a lightweight authentication middleware for Node.js that is extremely flexible and modular.
+Passport.js offers comprehensive set of link:http://passportjs.org/docs/configure[strategies] that support authentication
+using a username and password, Facebook, Twitter, and more.
+For more information see link:http://passportjs.org/[passport.org].
+
+- Can be bundled into existing node.js server
+- Available by default when running our reference server application
+- Covers authentication and authorization of express routes
+
+== Keycloak
+Keycloak is a full identity management solution with features that extend standard authorization and authentication scope.
+Keycloak will be suggested for solutions that require more comprehensive integration with company infrastructure and
```

Comment:
We recommend using Keycloak in Production to avail of the many features available for Enterprise Security including:

* [Single-Sign On](https://keycloak.gitbooks.io/documentation/server_admin/topics/sso-protocols.html)
* [User Federation](https://keycloak.gitbooks.io/documentation/content/server_admin/topics/user-federation.html)
* [Client Cert Authentication](https://keycloak.gitbooks.io/documentation/content/server_admin/topics/authentication/x509.html)
* [2FA with OTP](https://keycloak.gitbooks.io/documentation/content/server_admin/topics/authentication/otp-policies.html)
* [User Session Management](https://keycloak.gitbooks.io/documentation/content/server_admin/topics/sessions/administering.html)
* [Auditing](https://keycloak.gitbooks.io/documentation/server_admin/topics/events.html)
* [Brute Force Detection](https://keycloak.gitbooks.io/documentation/content/server_admin/topics/threat/brute-force.html)

:1st_place_medal:  This was one of the biggest things to really push in the docs and it was left out..... Customers seeing the above features could swing or sway using the entire solution based on the presence of these.

Reply:
had this in originally , @wtrocki taught it was too technical.  @jstafford  links to hard to maintain.  Will stick it back and let John decide.

Reply:
![](https://i.imgflip.com/1upa3d.jpg)

Reply:
@austincunningham The links are to the master Keycloak docs, rather than per-version docs so I can't see any maintenance from my point of view. So i think it should be ok.